AI/Nullkiller/Behaviors/BuyArmyBehavior.cpp

```diff
@@ -28,29 +28,45 @@ Goals::TGoalVec BuyArmyBehavior::decompose(const Nullkiller * ai) const
 {
 	Goals::TGoalVec tasks;
 
-	if(ai->cb->getDate(Date::DAY) == 1)
-		return tasks;
-		
 	auto heroes = cb->getHeroesInfo();
 
 	if(heroes.empty())
 	{
 		return tasks;
 	}
 
+	ai->dangerHitMap->updateHitMap();
+
 	for(auto town : cb->getTownsInfo())
 	{
+		//If we can recruit a hero that comes with more army than he costs, we are better off spending our gold on them
```

Comment:
You might also want to consider hero faction. Sure, necromancer might provide more troops than buying troops directly, but its undead troops would kill morale and require free slots in army to actually benefit from new troops

Reply:
I will remove this check entirely because it is pointless. I debugged at some point and found out that the army heroes come with usually is around 1k and thus never fulfills the criteria of having more value than if the army was bought directly.